src/probability/process/filtration.lean

```diff
@@ -211,6 +211,29 @@ lemma integrable.uniform_integrable_condexp_filtration
   uniform_integrable (λ i, μ[g | f i]) 1 μ :=
 hg.uniform_integrable_condexp f.le
 
+section of_set
+
+variables [preorder ι]
+
+/-- Given a sequence of measurable sets `(sₙ)`, `filtration_of_set` is the smallest filtration
+such that `sₙ` is measurable with respect to the `n`-the sub-σ-algebra in `filtration_of_set`. -/
+def filtration_of_set {s : ι → set Ω} (hsm : ∀ i, measurable_set (s i)) : filtration ι m :=
+{ seq := λ i, measurable_space.generate_from {t | ∃ j ≤ i, s j = t},
+  mono' := λ n m hnm, measurable_space.generate_from_mono
+    (λ t ⟨k, hk₁, hk₂⟩, ⟨k, hk₁.trans hnm, hk₂⟩),
+  le' := λ n, measurable_space.generate_from_le (λ t ⟨k, hk₁, hk₂⟩, hk₂ ▸ hsm k) }
```

Comment:
Could you check that this is the same filtration as the natural filtration of indicators of the sets? The way you construct this one is different enough from `filtration.natural` that I'd like to have that checked.
Also your `Indep_set.condexp_indicator_filtration_of_set_ae_eq` could then be seen as a consequence of a lemma about the condexp of a process at time (n+1) wrt its natural filtration at n, under an independence condition.

Reply:
Done!